plugin-base/src/main/java/io/everytrade/server/plugin/impl/everytrade/parser/exchange/binance/v4/BinanceSortedGroupV4.java

```diff
@@ -589,8 +593,46 @@ private void createBuySellTxs() {
             quoteRow = ndRow;
         }
 
-        var txsBuySell = new BinanceBeanV4();
-        txsBuySell.setDate(baseRow.getDate());
+        if (baseRow.getOperationType().code.equals("ETH 2.0 STAKING")) {
+            for (int i = 0; i < rowBuySellRelated.size(); i++) {
+                var txsBuySell = newBinanceBeanV4(stRow, baseRow, quoteRow, relatedTransaction);
+                if (i == 1 && baseRow.getCoin().code().equals("BETH") && quoteRow.getCoin().code().equals("ETH")){
```

Comment:
Do not use String of Currencies but ENUMs. The conditions should not contain BETH currency.